Source/JavaScriptCore/runtime/WriteBarrier.h

```diff
@@ -352,3 +352,21 @@ class WriteBarrierStructureID {
 };
 
 } // namespace JSC
+
+namespace WTF {
+
+template<typename T> struct VectorTraits<JSC::WriteBarrier<T>> : public SimpleClassVectorTraits {
+    static_assert(std::is_trivially_destructible<JSC::WriteBarrier<T>>::value);
+    static constexpr bool canCopyWithMemcpy = true;
```

Comment:
@Constellation Isn't this wrong? My understanding is that memcpy is free to use load/store instructions with fewer than 8 bytes, which would cause GC tearing. I know we do something special for some copies, what is the current state of that?